test/project_types/script/layers/infrastructure/languages/project_creator_test.rb

```diff
@@ -36,4 +53,31 @@
       end
     end
   end
+
+  describe "setup_dependencies" do
+    subject { project_creator.setup_dependencies }
+
+    it "should setup dependencies" do
+      # setup_sparse_checkout
+      ShopifyCli::Git
+        .expects(:sparse_checkout)
+        .with(
+          extension_point.sdks.assemblyscript.repo,
+          project_creator.sparse_checkout_set_path,
+          branch,
+          context
+        )
+        .once
+
+      # clean
+      source = File.join(project_creator.path_to_project, project_creator.sparse_checkout_set_path)
+      FileUtils.expects(:copy_entry).with(source, project_creator.path_to_project)
+
+      # set_script_name
+      File.expects(:read).with("package.json").returns("name = #{type.gsub("_", "-")}-default")
+      File.expects(:write).with("package.json", "name = #{script_name}")
```

Comment:
Please look into the use of `FakeFS`.  You can find examples in other unit tests.

Reply:
I don't think this is the correct way of using FakeFS.  I realized that perhaps what's in our code base was wrong.  Here is the doc: https://github.com/fakefs/fakefs.